pkg/scalers/solr_scaler.go

```diff
@@ -24,16 +23,42 @@ type solrScaler struct {
 }
 
 type solrMetadata struct {
-	host                       string
-	collection                 string
-	targetQueryValue           float64
-	activationTargetQueryValue float64
-	query                      string
-	triggerIndex               int
+	triggerIndex int
+
+	Host                       string  `keda:"name=host, order=triggerMetadata"`
+	Collection                 string  `keda:"name=collection, order=triggerMetadata, optional"`
+	TargetQueryValue           float64 `keda:"name=targetQueryValue, order=triggerMetadata"`
+	ActivationTargetQueryValue float64 `keda:"name=activationTargetQueryValue, order=triggerMetadata, optional, default=0"`
+	Query                      string  `keda:"name=query, order=triggerMetadata, optional"`
 
 	// Authentication
-	username string
-	password string
+	Username string `keda:"name=username, order=authParams;triggerMetadata"`
+	Password string `keda:"name=password, order=authParams;triggerMetadata"`
+}
+
+func (s *solrMetadata) Validate() error {
+	if s.Host == "" {
+		return fmt.Errorf("no host given")
+	}
+
+	if s.Collection == "" {
+		return fmt.Errorf("no collection given")
+	}
+
+	if s.Query == "" {
+		s.Query = "*:*"
+	}
+
+	// Parse Authentication
+	if s.Username == "" {
+		return fmt.Errorf("no username given")
+	}
+
+	if s.Password == "" {
+		return fmt.Errorf("no password given")
+	}
+
+	return nil
```

Comment:
nit: this can be made a bit simpler thanks to the `optional` tags in the struct
```suggestion
func (s *solrMetadata) Validate() error {
	if s.Query == "" {
		s.Query = "*:*"
	}
	return nil
```